### PR TITLE
feat: discussions

### LINKED ISSUE #385: List of subgraphs in composition detail screen is not scrollable
### Component(s)

studio

## Description

In composition detail screen having lot of subgraphs I am not able to scroll through all of them.

Screenshot 2023-12-19 at 11 19 27

### Component version

v0.44.1

### Router configuration

_No response_

### Router execution config

_No response_

### Log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by thisisnithin**
## Motivation and Context

- Have multiple discussions across schema versions and graphs
- Reply/Edit/Delete comments in the discussion thread
- Mark discussion as resolved
- Show/Hide Discussions on schema
- Two way linking between explorer and SDL

Future improvements
- Notifications through email
- Mentions
- Moving discussions across graphs

## Issues

Completes ENG-4644
Fixes #385  
Fixes ENG-4724

## Preview
image
image
image
image

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Review by jensneuse — APPROVED** [review]

File: controlplane/src/db/schema.ts [hunk 1]
```diff
@@ -1012,3 +1012,58 @@ export const subgraphMembers = pgTable(
...
+
+export const discussionThread = pgTable('discussion_thread', {
+  id: uuid('id').notNull().primaryKey().defaultRandom(),
+  discussionId: uuid('discussion_id')
+    .notNull()
+    .references(() => discussions.id, {
+      onDelete: 'cascade',
+    }),
+  contentMarkdown: text('content_markdown'),
+  contentJson: customJson('content_json'),
```
Comment (StarpTech) [inline controlplane/src/db/schema.ts:1041]: You can specify a type here to be type safe during development.

File: studio/src/hooks/use-window-size.ts [hunk 2]
```diff
@@ -32,6 +32,7 @@ export default function useWindowSize() {
   return {
     windowSize,
     isMobile: typeof windowSize?.width === "number" && windowSize?.width < 768,
+    isTablet: typeof windowSize?.width === "number" && windowSize?.width < 1024,
```
Comment (StarpTech) [inline studio/src/hooks/use-window-size.ts:35]: Is that aligned with tailwind?

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: controlplane/src/db/schema.ts (hunk 1 above)
Comment (thisisnithin) [reply controlplane/src/db/schema.ts]: Done

File: studio/src/hooks/use-window-size.ts (hunk 2 above)
Comment (thisisnithin) [reply studio/src/hooks/use-window-size.ts]: Yes, but we don't need it, cleaned it up

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**